### DISCUSSION TITLE
Is it possible to use GitHub Actions to manage an open source project? ie Fork PRs fail

### DISCUSSION THREAD

**Discussion by benjamincharity** [community/community#25501] [Actions]
<p>I feel silly even typing that title as I’m <em>sure</em> I’ve missed something…</p>
<p>We have an open source project that is using GitHub Actions as the CI/CD and all has been working great for our team members.</p>
<p>But when someone outside of our organization forks our repo and submits a PR, our GHA fails due to missing tokens. I do understand that care must be taken to not expose secrets outside of the organization, but I also must believe that this is possible somehow, or GHA cannot be a viable CI/CD solution for open source projects.</p>
<p>I had thought that <code>pull_request_target</code> may solve that very issue, but we haven’t had luck there either.</p>
<p>Any info or pointers would be greatly appreciated :pray:</p>

**Comment by Simran-B** [comment]
<p>What exactly fails? Can you share the workflow?</p>

**Comment by benjamincharity** [comment]
<p>Sorry - I could have made my initial post much better.</p>
<p>When someone forks and creates a PR, it fails with a missing token: <a href="https://github.com/GetTerminus/terminus-oss/runs/1177183898?check_suite_focus=true#step:5:11" rel="noopener nofollow ugc">https://github.com/GetTerminus/terminus-oss/runs/1177183898?check_suite_focus=true#step:5:11</a></p>
<p>We tried using the newer <code>pull_request_target</code> but then the workflow is just skipped with no errors or messages: <a href="https://github.com/GetTerminus/terminus-oss/actions/runs/279005642" rel="noopener nofollow ugc">https://github.com/GetTerminus/terminus-oss/actions/runs/279005642</a></p>

**Comment by otter-computer** [comment, marked as answer]
<p>It looks like you’re setting the <code>GITHUB_TOKEN</code> env var using <code>${{ secrets.GH_TOKEN }}</code> instead of <code>${{ secrets.GITHUB_TOKEN }}</code>. Did you manually create this secret by passing in a personal access token? If so, is there a specific reason you’ve done this—i.e. are specific permissions required that aren’t provided by the regular token?</p>
<p>Secrets aren’t passed to workflows that run on forks so this is likely why this variable is turning up empty:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow" target="_blank" rel="noopener">Encrypted secrets - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>The only secret that is passed on to fork workflows is a read-only version of the <code>GITHUB_TOKEN</code>, so if you can switch to using that this workflow should run. Otherwise it would be good to hear what permissions this token needs in order to execute your workflow.</p>

**Comment by benjamincharity** [comment]
<p>That is a great point. I remember needing to make that switch at <em>some</em> point when originally migrating to GHA, but it’s very possible it was needed for our release flow vs pr etc.</p>
<p>I just tried switching back and now see a branch error - so it seems the token was the issue. I’ll keep exploring my other issues.</p>
<p>Thanks for pointing me in the correct direction <a class="mention" href="/thomasshaped">@thomasshaped</a></p>
